tests/test.linter.js

```diff
@@ -233,82 +223,73 @@ describe('Linter', () => {
+  it('should throw when message.type is undefined', async () => {
     const addonLinter = new Linter({ _: ['tests/fixtures/webextension.zip'] });
     addonLinter.io = { files: { whatever: {} } };
     addonLinter.io.getFile = () => Promise.resolve();
     addonLinter.getScanner = sinon.stub();
     class FakeScanner {
-      scan() {
-        return Promise.resolve({
+      async scan() {
+        return {
           linterMessages: [{ message: 'whatever' }],
           scannedFiles: [],
-        });
+        };
       }
     }
     addonLinter.getScanner.returns(FakeScanner);
-    return addonLinter.scanFile('whatever')
-      .then(unexpectedSuccess)
-      .catch((err) => {
-        expect(err.message).toContain('message.type must be defined');
-      });
+    await expect(
+      addonLinter.scanFile('whatever')
+    ).rejects.toThrow('message.type must be defined');
   });
 
 
-  it('should see an error if scanFiles() blows up', () => {
+  it('should see an error if scanFiles() blows up', async () => {
     const addonLinter = new Linter({ _: ['foo'] });
     addonLinter.checkFileExists = fakeCheckFileExists;
     // Stub handleError to prevent output.
     addonLinter.handleError = sinon.stub();
-    addonLinter.scanFiles = () => {
-      return Promise.reject(new Error('scanFiles explosion'));
+    addonLinter.scanFiles = async () => {
+      throw new Error('scanFiles explosion');
     };
 
     class FakeXpi extends FakeIOBase {
-      getFilesByExt() {
-        return Promise.resolve(['foo.js', 'bar.js']);
+      async getFilesByExt() {
+        return ['foo.js', 'bar.js'];
       }
     }
 
-    return addonLinter.scan({ _Xpi: FakeXpi })
-      .then(unexpectedSuccess)
-      .catch((err) => {
-        expect(err).toBeInstanceOf(Error);
-        expect(err.message).toContain('scanFiles explosion');
-      });
+    await expect(
+      addonLinter.scan({ _Xpi: FakeXpi })
+    ).rejects.toThrow('scanFiles explosion');
   });
 
-  it('should call addError when Xpi rejects with dupe entry', () => {
+  it('should call addError when Xpi rejects with dupe entry', async () => {
     const addonLinter = new Linter({ _: ['bar'] });
     addonLinter.checkFileExists = fakeCheckFileExists;
     addonLinter.collector.addError = sinon.stub();
     addonLinter.print = sinon.stub();
     class FakeXpi extends FakeIOBase {
-      getFiles() {
-        return Promise.reject(
-          new Error('DuplicateZipEntry the zip has dupes!'));
+      async getFiles() {
+        throw new Error('DuplicateZipEntry the zip has dupes!');
       }
       getFilesByExt() {
         return this.getMetadata();
       }
     }
-    return addonLinter.scan({ _Xpi: FakeXpi })
-      .then(unexpectedSuccess)
-      .catch(() => {
-        sinon.assert.calledWith(
-          addonLinter.collector.addError,
-          messages.DUPLICATE_XPI_ENTRY);
-        sinon.assert.calledOnce(addonLinter.print);
-      });
+    try {
+      await addonLinter.scan({ _Xpi: FakeXpi });
```

Comment:
@hameleonka let's try to rewrite this using the `await expect(...).rejects...` jest assertion.